source/common/unified_memory_allocation/src/memory_provider.c

```diff
@@ -6,14 +6,15 @@
  *
  */
 
+#include "memory_tracker_internal.h"
 #include <uma/memory_provider.h>
 
 #include <assert.h>
 #include <stdlib.h>
 
 struct uma_memory_provider_t {
+    void *priv;
```

Comment:
It looks like a simple refactoring, isn't it? If so, I think it is better to do that in a separate PR.